### ISSUE TITLE
Brotli Support?

### ISSUE THREAD

**Issue by gustavneustadt** [tsolomko/SWCompression#36]
Are there plans for [Brotli](https://github.com/google/brotli) Support?
Or are there compatible formats or other ways to use Brotli in Swift?

Thanks. Keep up the great work!

**Comment by tsolomko** [comment]
> Are there plans for [Brotli](https://github.com/google/brotli) Support?
> [This](https://github.com/foliojs/brotli.js) or [that](https://pub.dev/documentation/es_compression/latest/brotli/brotli-library.html) may be helpful.

No, unfortunately, at the moment there are no such plans, partially because I was not aware of Brotli existence. I will add its support to the ever-growing todo list. 

> Or are there compatible formats or other ways to use Brotli in Swift?
>
> Thanks. Keep up the great work!

While, from what I understand, algorithmically Brotli consists of commonly used parts such as LZ77 and Huffman, since the metadata format is unique, I don't think there is any compatibility with other formats. I am also not aware of any other implementations of Brotli in Swift.

Thank for your kind words and your suggestion, though.

**Comment by gustavneustadt** [comment]
In my current project I now use [matthewcheok/Brotli](https://github.com/matthewcheok/Brotli). Maybe its worth a look.

Brotli is used in woff2 font format. Seems like it was designed for the web applications. May be it gets implemented in the future in more and more data formats.
